pkg/storage/ingest/reader_client.go

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ingest
+
+import (
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/twmb/franz-go/pkg/kgo"
+	"github.com/twmb/franz-go/plugin/kprom"
+)
+
+// NewKafkaReaderClient returns the kgo.Client that should be used by the Reader.
+func NewKafkaReaderClient(cfg KafkaConfig, metrics *kprom.Metrics, logger log.Logger, opts ...kgo.Opt) (*kgo.Client, error) {
+	const fetchMaxBytes = 100_000_000
+
+	opts = append(opts, commonKafkaClientOptions(cfg, metrics, logger)...)
+	opts = append(opts,
+		kgo.FetchMinBytes(1),
+		kgo.FetchMaxBytes(fetchMaxBytes),
+		kgo.FetchMaxWait(5*time.Second),
+		kgo.FetchMaxPartitionBytes(50_000_000),
+
+		// BrokerMaxReadBytes sets the maximum response size that can be read from
+		// Kafka. This is a safety measure to avoid OOMing on invalid responses.
+		// franz-go recommendation is to set it 2x FetchMaxBytes.
+		kgo.BrokerMaxReadBytes(2*fetchMaxBytes),
+	)
+	client, err := kgo.NewClient(opts...)
+	if err != nil {
+		return nil, errors.Wrap(err, "creating kafka client")
+	}
+
+	return client, nil
+}
+
+func NewKafkaReaderClientMetrics(component string, reg prometheus.Registerer) *kprom.Metrics {
+	return kprom.NewMetrics("cortex_ingest_storage_reader",
+		kprom.Registerer(prometheus.WrapRegistererWith(prometheus.Labels{"component": component}, reg)),
```

Comment:
Note to reviewers: I've added the "component" label here, in order to not have conflicting metrics in single binary mode (on the other side, we never tested ingest storage in single binary so other stuff may not work at all).

Reply:
we used to have the partition label here, but I don't think it ever made much sense because there's only one partition per consumer. Agree? 

This might break the block builder, but provided it's still not merged maybe we shouldn't worry about it

Reply:
> we used to have the partition label here, but I don't think it ever made much sense because there's only one partition per consumer. Agree?

Agree. I commented about it here: https://github.com/grafana/mimir/pull/8808/files#r1690885330